components/bl-color-picker-component/src/index.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { PickerContainer } from './picker-container';
+import { PickerTrigger } from './picker-trigger';
+
+export default function ColorPickerComponent({ component, eventHandlers }) {
+  const { display, classList, pickerTriggerVisibility } = component;
+  const { onChangeColor } = eventHandlers;
+
+  const pickerContainerRef = useRef(null);
+  const [currentColor, setCurrentColor] = useState('');
+  const [pickerVisibility, setPickerVisibility] = useState(!pickerTriggerVisibility);
+
+  const handleClickOutsidePicker = e => {
+    if (!pickerContainerRef.current.contains(e.target)) {
+      setPickerVisibility(false);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
yes, we are going to open a list of public hooks, however, it will be available after implementing ComponentPods, so let's do not wait it in the component